### ISSUE TITLE
Image Resizing

### ISSUE THREAD

**Issue by 4leyam** [SDWebImage/SDWebImageSwiftUI#111]
hi, i was wondering if there is a way to resize a fetched image (Real image not the frame) and cache it using this library cause it seems like even if the image frame is set to a certain dimension, the image itself keeps it size.
Sorry if this question has already been answered elsewhere but i couldn't find any useful information.
To work around it i'm using the following code
`

 self.imageManager.setOnSuccess { (image , cache) in
                    
                    DispatchQueue.global(qos: .userInitiated).async {
                     
                        if let cgImage = image.cgImage {
                            
                            let dim : Int = 200
                            
                            let thumbImage : UIImage = image.sd_resizedImage(with: CGSize(width: dim, height: dim), scaleMode: .aspectFill) ?? UIImage(cgImage: cgImage)
                            
                            if Video.cachedT.count >= 20 {
                                
                                var tmp = Video.cachedT.sorted { (arg0 : (key: Int, value: String), arg1 : (key: Int, value: String)) -> Bool in
                                    
                                    arg0.key >= arg1.key
                                    
                                }
                                tmp.removeLast()
                                
                            }
                             
                            //then cache the image inside an array
                            //and update the view
                            self.loaded = true
                        }
                        
                    }
                    
                }
                
                self.imageManager.load()

`
In summary, i'm asking if there is a way for me not to write this code by myself using your library?
if not, is there any way i can give back the resized image so that the library can handle the caching step

thank you in advance for your help

**Comment by dreampiggy** [comment]
Using the SDWebImage's transformer. See wiki: https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#image-transformer-50

```swift
static let transformer = SDImageResizingTransformer(size: .init(width: 100, height: 100), scaleMode: .fit)

WebImage(url: url, context: [.imageTransformer: transformer])
```

Which do the right thing for this.

Using the `onSuccess` does not always works, because think this case: You transform a image into a smaller one, but, you need to:

+ Update the Cache (or it keep full size)
+ You need another cache key (or it will override the full size image)
+ You need to dispatch to global queue do image resizing and callback in main queue for UI (or this cause render issue)
+ You need to keep the `memory cache safe`, because for memory cache, we assume it should be immediately queried, not during any async operation.

So, just use SDWebImage's transformer feature, which save you from all this problem.

**Comment by dreampiggy** [comment]
And also, If you means, to show a `Thumbnail` (This is a subset for any transformer, only about the image size). You can also use thumbnail decoding.

The thumbnail decoding is better than transformer, because it does not allocate any full image buffer size. For example, if you want to display a 40000*40000 pixels image, which will consume 6GB RAM of your iPhone and your App will crash immediatlly, even you use resize transformer. Because that need to decode full image firstlly and scale down.

Use thumbnail decoding will directlly only decode thumbnailed size, without full pixels allocation.

See more in wiki: https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#thumbnail-decoding-550

**Comment by 4leyam** [comment]
Oooohh ok, i haven’t seen it before, thanks a million @dreampiggy
